### PR TITLE
entgql: use goModel directive for mapping

### PR DESCRIPTION
**Pull request by a8m**
In case the GraphQL node name is different from the Ent name, `autobind` won't work, and we need to map it explicitly.

### CODE REVIEW

File: entgql/schema.go [hunk 1]
```diff
@@ -24,12 +24,60 @@ import (
...
+	nodes     []*gen.Type
+	relaySpec bool
+}
+
+var (
+	pos        = &ast.Position{Src: &ast.Source{BuiltIn: false}}
+	directives = map[string]*ast.DirectiveDefinition{
+		"goModel": {
+			Name:     "goModel",
+			Position: pos,
```
Comment (giautm) [inline entgql/schema.go:39]: position can be set to nil

Comment (a8m) [reply entgql/schema.go:39]: It's needed here, but not in other places. 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x150c2c2]

goroutine 1 [running]:
github.com/vektah/gqlparser/v2/formatter.(*formatter).FormatDirectiveDefinition(0x15a7f60, 0xc000347a40)
```

https://github.com/vektah/gqlparser/blob/master/formatter/formatter.go#L326-L328

**Review by giautm — APPROVED** [review]
LGTM

**Comment by giautm** [comment]
How to mapping the custom type CategoryConfig to GoType with the goModel directive?

**Comment by a8m** [comment]
> How to mapping the custom type CategoryConfig to GoType with the goModel directive?

I'll map fields with custom GoType in a different PR.

**Pull request merged**